lib/common/notifications.js

```diff
@@ -17,6 +17,22 @@ var _validateDocument = function(notification) {
     badge: Match.Optional(Number),
     sound: Match.Optional(String),
     notId: Match.Optional(Match.Integer),
+    apn: Match.Optional({
+      from: Match.Optional(String),
+      title: Match.Optional(String),
+      text: Match.Optional(String),
+      badge: Match.Optional(Number),
+      sound: Match.Optional(String),
+      notId: Match.Optional(Match.Integer)
+    }),
+    gcm: Match.Optional({
+      from: Match.Optional(String),
+      title: Match.Optional(String),
+      text: Match.Optional(String),
+      badge: Match.Optional(Number),
+      sound: Match.Optional(String),
+      notId: Match.Optional(Match.Integer)
+    }),
```

Comment:
Yes, I think that the readability is important and maybe we can have different options for APN and GCM.